src/Compilers/Core/Portable/ConstantValueSpecialized.cs

```diff
@@ -660,6 +692,22 @@ public override uint UInt32Value
                 }
             }
 
+            public override long Int64Value
+            {
+                get
+                {
+                    switch (this.Discriminator)
```

Comment:
What does Discriminator track for ConstantValue?  There is no comment and reading the code it's hard to understand why only the values specified here would be valued for each of the new overrides.